src/test/scala/chiselTests/BundleLiterals.scala

```diff
@@ -0,0 +1,66 @@
+// See LICENSE for license details.
+
+package chiselTests
+
+import chisel3._
+import org.scalatest._
+
+class BundleWithLit(lit: Option[Int]=None) extends Bundle {
+  val internal = lit.map(_.U).getOrElse(UInt(4.W))
+  override def cloneType = new BundleWithLit(lit).asInstanceOf[this.type]
+}
+
+object BundleWithLit {
+  def apply(x: Int) = {
+    val ret = new BundleWithLit(Some(x))
+    Aggregate.LitBind(ret)
+    ret
+  }
+  def apply() = {
+    new BundleWithLit()
+  }
+}
+
+class BundleWithBundleWithLit(lit: Option[Int]=None) extends Bundle {
+  val internal = lit.map(BundleWithLit(_)).getOrElse(BundleWithLit())
+  override def cloneType = new BundleWithBundleWithLit(lit).asInstanceOf[this.type]
+}
+
+object BundleWithBundleWithLit {
+  def apply(x: Int) = {
+    val ret = new BundleWithBundleWithLit(Some(x))
+    Aggregate.LitBind(ret)
```

Comment:
Seems strange to have to call this in user code; why is it here?

Reply:
This is patterned after `UIntFactory`, which isn't user code b/c you don't subtype `UInt`, but users do subtype bundles. I suppose you could have an aggregate try to detect if its members are literals, but that seemed to be a bigger change so I didn't take that step.